tests/Dfc.CourseDirectory.WebV2.Tests/Security/SyncUserProviderSignInActionTests.cs

```diff
@@ -0,0 +1,276 @@
+﻿using Dfc.CourseDirectory.WebV2.DataStore.CosmosDb;
+using Dfc.CourseDirectory.WebV2.DataStore.CosmosDb.Models;
+using Dfc.CourseDirectory.WebV2.DataStore.CosmosDb.Queries;
+using Dfc.CourseDirectory.WebV2.Helpers;
+using Dfc.CourseDirectory.WebV2.Helpers.Interfaces;
+using Dfc.CourseDirectory.WebV2.Security;
+using Dfc.CourseDirectory.WebV2.Services.Interfaces;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using UkrlpService;
+using Xunit;
+
+namespace Dfc.CourseDirectory.WebV2.Tests.Security
+{
+    public class SyncUserProviderSignInActionTests : MvcTestBase
+    {
+        private SyncUserProviderSignInAction signInActionToTest;
+        private IUkrlpSyncHelper _ukrlpSyncHelper;
+
+        public SyncUserProviderSignInActionTests(CourseDirectoryApplicationFactory factory)
+            : base(factory)
+        {
+        }
+
+        [Fact]
+        public async Task GivenSignInContext_WhenUkprnSuppliedAndProviderExists_EnsureProviderDataIsUpdated()
+        {
+            // Arrange
+            int providerUkprn = 01234566;
+            var providerData = this.GenerateProviderData(providerUkprn);
+
+            var createProviderContacts = new List<CreateProviderContact>();
+            providerData.ProviderContact.ToList().ForEach(c => createProviderContacts.Add(new CreateProviderContact()
+            {
+                ContactType = c.ContactType,
+                ContactTelephone1 = c.ContactTelephone1,
+                ContactWebsiteAddress = c.ContactWebsiteAddress,
+                ContactEmail = c.ContactEmail,
+                AddressSaonDescription = c.ContactAddress.SAON?.Description,
+                AddressPaonDescription = c.ContactAddress.PAON?.Description,
+                AddressStreetDescription = c.ContactAddress.StreetDescription,
+                AddressLocality = c.ContactAddress.Locality,
+                AddressItems = c.ContactAddress.Items,
+                AddressPostCode = c.ContactAddress.PostCode,
+                PersonalDetailsGivenName = c.ContactPersonalDetails?.PersonGivenName.FirstOrDefault(),
+                PersonalDetailsFamilyName = c.ContactPersonalDetails?.PersonFamilyName,
+            }));
+
+            providerData.Id = await TestData.CreateProvider(providerUkprn,
+                                                            providerData.ProviderName,
+                                                            providerData.ProviderType,
+                                                            providerData.ProviderStatus,
+                                                            Models.ApprenticeshipQAStatus.Passed,
```

Comment:
I'd omit `apprenticeshipQAStatus`, `marketingInformation`, `courseDirectoryName` and `alias` here since they're not relevent to this test